samples/OpenIdConnectSample/Startup.cs

```diff
@@ -31,28 +38,63 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
             });
 
             app.UseOpenIdConnectAuthentication(options =>
+            {
+                options.ClientId = "fe78e0b4-6fe7-47e6-812c-fb75cee266a4";
+                options.Authority = "http://localhost:12345";
+                options.RedirectUri = "http://localhost:12345";
+                options.UsePostAuthorizationRequests = true;
+            });
+
+            app.UseOpenIdConnectServer(options =>
+            {
+                options.AllowInsecureHttp = true;
+
+                options.UseCertificate(
+                    assembly: typeof(Startup).GetTypeInfo().Assembly,
+                    resource: "OpenIdConnectSample.Certificate.pfx",
+                    password: "Owin.Security.OpenIdConnect.Server");
+
+                options.Provider = new OpenIdConnectServerProvider
                 {
-                    options.ClientId = "fe78e0b4-6fe7-47e6-812c-fb75cee266a4";
-                    options.Authority = "https://login.windows.net/cyrano.onmicrosoft.com";
-                    options.RedirectUri = "http://localhost:42023";
-                });
+                    OnValidateClientAuthentication = notification =>
+                    {
+                        notification.Validated();
+                        return Task.FromResult<object>(null);
+                    },
+
+                    OnValidateClientRedirectUri = notification =>
+                    {
+                        notification.Validated();
+                        return Task.FromResult<object>(null);
+                    },
+
+                    OnAuthorizationEndpoint = async notification =>
+                    {
+                        var identity = new ClaimsIdentity(new[]
+                        {
+                            new Claim("unique_name", "Bob").WithDestination("id_token"),
+                            new Claim(ClaimTypes.NameIdentifier, "7b1d94bc-0f83-4035-99f4-aa9ccd30a1f3")
+                        }, notification.Options.AuthenticationScheme);
+
+                        var principal = new ClaimsPrincipal(identity);
+
+                        await notification.HttpContext.Authentication.SignInAsync(notification.Options.AuthenticationScheme, principal);
+                        notification.HandleResponse();
+                    }
+                };
+            });
 
             app.Run(async context =>
             {
                 if (!context.User.Identities.Any(identity => identity.IsAuthenticated))
                 {
                     await context.Authentication.ChallengeAsync(OpenIdConnectAuthenticationDefaults.AuthenticationScheme, new AuthenticationProperties { RedirectUri = "/" });
-
-                    context.Response.ContentType = "text/plain";
```

Comment:
Updating the headers collection at this stage is no longer safe with POST requests since we start writing the auto-post form to the response stream in the middleware (invoked via `context.Authentication.ChallengeAsync`).

These two lines were pointless anyway, since the user agent was not supposed to see the printed text.